s3direct/views.py

```
@@ -60,4 +61,7 @@ def get_upload_params(request):
         server_side_encryption
     )
 
+    if hasattr(run_after, '__call__'):
```

Unlike key, this will only work with callables. This should be:

```python
if run_after is not None:
    run_after(request)
```

The code, as you have it, will silently ignore non-callable settings.